tests/utils/tools.php

```diff
@@ -220,6 +220,35 @@ function is_mongos($uri)
     return get_primary_server($uri)->getType() === Server::TYPE_MONGOS;
 }
 
+/**
+ * Checks that the topology is a sharded cluster using a replica set
+ */
+function is_mongos_with_replica_set($uri)
```

Comment:
Ripped out of PHPLIB, eh? :smile:

Reply:
Your code was so good I couldn't bring myself to rewriting it 😁